src/libraries/System.Text.Json/Common/JsonSimpleNamingPolicy.cs

```diff
@@ -0,0 +1,147 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Buffers;
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal sealed class JsonSimpleNamingPolicy : JsonNamingPolicy
+    {
+        private readonly bool _lowercase;
+        private readonly char _boundary;
+
+        internal JsonSimpleNamingPolicy(bool lowercase, char boundary) =>
+            (_lowercase, _boundary) = (lowercase, boundary);
+
+        public override string ConvertName(string name)
+        {
+            int bufferLength = name.Length * 2;
+            char[]? buffer = bufferLength > 512
+                ? ArrayPool<char>.Shared.Rent(bufferLength)
+                : null;
+
+            int resultLength = 0;
+            Span<char> result = buffer is null
+                ? stackalloc char[512]
+                : buffer;
+
+            void WriteWord(ref Span<char> result, ReadOnlySpan<char> word)
+            {
+                if (word.IsEmpty)
+                    return;
+
+                int required = result.IsEmpty
+                    ? word.Length
+                    : word.Length + 1;
+
+                if (required >= result.Length)
+                {
+                    int bufferLength = result.Length * 2;
+                    char[] bufferNew = ArrayPool<char>.Shared.Rent(bufferLength);
+
+                    result.CopyTo(bufferNew);
+
+                    if (buffer is not null)
+                        ArrayPool<char>.Shared.Return(buffer);
+
+                    buffer = bufferNew;
+                }
+
+                if (resultLength != 0)
+                {
+                    result[resultLength] = _boundary;
+                    resultLength += 1;
+                }
+
+                Span<char> destination = result.Slice(resultLength);
```

Comment:
Nit/code style - you could use the slicing language feature.

something like:
```suggestion
                Span<char> destination = result[..resultLength];
```

Reply:
Originally I used them, but it caused compilation issues, so have to use methods. Maybe something changed since that I can try indexers, let's see.

Reply:
You can use sharplab.io to confirm whether they get compiled identically/similar ;)

Reply:
Unfortunately Roslyn doesn't emit ideal code for the range / slices feature. So if perf is critical the method-approach is "better".